src/check_datapackage/check.py

```diff
@@ -254,11 +262,54 @@ def _handle_S_resources_x_path(
     return edits
 
 
+def _handle_S_resources_x_schema_fields_x(
+    parent_error: SchemaError,
+    schema_errors: list[SchemaError],
+) -> SchemaErrorEdits:
+    """Only flag errors for the relevant field type.
+
+    E.g., if the field type is `string`, flag errors for the string-based schema only.
+    """
+    edits = SchemaErrorEdits()
+    errors_in_group = _get_errors_in_group(schema_errors, parent_error)
+    edits.remove.append(parent_error)
+
+    field_type: str = parent_error.instance.get("type", "string")
+
+    # The field's type is unknown
+    if field_type not in FIELD_TYPES:
+        unknown_field_error = SchemaError(
+            message=(
+                "The type property in this resource schema field is incorrect. "
+                f"The value can only be one of these types: {', '.join(FIELD_TYPES)}."
```

Comment:
Is that a missing `}`? Should it be:

```suggestion
                f"The value can only be one of these types: {', '.join(FIELD_TYPES)', '}.'"
```